src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -156,15 +157,65 @@ private static IServiceCollection TrySetProductInfo(this IServiceCollection serv
             return services.Configure<ServiceManagerOptions>(o => o.ProductInfo ??= productInfo);
         }
 
-        private static IServiceCollection AddRestClientFactory(this IServiceCollection services) => services
-            .AddHttpClient(Options.DefaultName, (sp, client) => client.Timeout = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value.HttpClientTimeout)
-            .ConfigurePrimaryHttpMessageHandler(sp => new HttpClientHandler() { Proxy = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value.Proxy }).Services
-            .AddSingleton(sp =>
+        private static IServiceCollection AddRestClientFactory(this IServiceCollection services)
+        {
+            // For AAD, health check.
+            services
+                .AddHttpClient(Options.DefaultName, (sp, client) => client.Timeout = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value.HttpClientTimeout)
+                .ConfigurePrimaryHttpMessageHandler(ConfigureProxy);
+
+            // For other data plane APIs.
+            services.AddSingleton<IBackOffPolicy>(sp =>
+            {
+                var options = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value;
+                var retryOptions = options.RetryOptions;
+                return retryOptions == null
+                    ? new DummyBackOffPolicy()
+                    : retryOptions.Mode switch
+                    {
+                        RetryMode.Fixed => ActivatorUtilities.CreateInstance<FixedBackOffPolicy>(sp),
+                        RetryMode.Exponential => ActivatorUtilities.CreateInstance<ExponentialBackOffPolicy>(sp),
+                        _ => throw new NotSupportedException($"Retry mode {retryOptions.Mode} is not supported.")
+                    };
+            });
+            services
+                .AddHttpClient(Constants.HttpClientNames.Resilient, (sp, client) =>
+                {
+                    var options = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value;
+                    if (options.RetryOptions == null)
+                    {
+                        client.Timeout = options.HttpClientTimeout;
+                    }
+                    // Else the timeout is enforced by TimeoutHttpMessageHandler.
```

Comment:
question: why not always use TimeoutHttpMessageHandler?

Reply:
To keep backward-compatibility. According to [HttpClient API doc](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=net-7.0), in case of timeout, different exceptions are thrown on different .NET implementations. I don't want to imitate all the implementations, so just let the original `HttpClient` to handle the timeout if no RetryOptions is set.